File: get_images.js
const gm = require('gm');
const fs = require('fs')
const pages = require('./_data/pages')
fs.mkdirSync('images')

pages.forEach(page => {
    let num = pad(page.emory_page, 4)
    let pagefile = `00825376_1523_${num}.jp2`
    console.log(pagefile)
    gm(`../bdb_raw/${pagefile}`)
        .resize(1024)
        .write(`images/${page.section}_${page.id}.png`, err => {
        if (err) console.log(err);
    });
})

function pad(int, total_len){
    var s = int+"";
    while (s.length < total_len) s = "0" + s;
    return s;
}

File: get_images.py
import wget
import json

with open('_data/pages.json', 'r') as pages_file:
    pages = json.load(pages_file)

    for page in pages:
        emory = page['emory_page']
        section = page['section']
        pid = page['id']
        url = f'http://iiif.archivelab.org/iiif/00825376.1523.emory.edu${emory}/full/1024/0/gray.png'
        print(f'\nDownloading: {section}: {pid} @ \n\t{url}')
        wget.download(url, f'images/{section}_{pid}.png')